gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/VerificationOnboardService.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.conformance;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * service class offered methods for checking onboarding information and also retrieve metadata from
+ * provided serviceid.
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class VerificationOnboardService {
+
+
+    private final DiscoveryConfigUri discoveryConfigUri;
+
+    @Qualifier("secureHttpClientWithoutKeystore")
+    private final CloseableHttpClient closeableHttpClient;
+
+    /**
+     * Accept serviceId and check if the service is onboarded to the API Mediation Layer
+     * @param serviceId accept serviceId to check
+     * @return return true if the service is known by Eureka otherwise false.
+     * @throws IOException
+     */
+    public boolean checkOnboarding(String serviceId) throws IOException {
+
+        Boolean check = false;
+        HttpGet httpget = constructHttpGet(serviceId);
+        try {
+            CloseableHttpResponse response = closeableHttpClient.execute(httpget);
```

Comment:
What is the reason for using httpclient here? I would suggest to use DiscoveryClient. Similar functionality can be found in Providers class.

Reply:
Thanks for reviewing. but I am not sure which DiscoveryClient should I use. Do you mean the DiscoveryClient in the netflix eureka class? Or the one DefaultEurekaClientConfigProvider in the providers class?

Reply:
Here is the example of how the gateway is currently getting info about zOSMF instance, it should be possible to achieve the same by injecting DiscoveryClient into your class and then passing serviceId to getInstance() method. https://github.com/zowe/api-layer/blob/c8355a2ed9d2562f2568c58ce74093e1064a611b/gateway-service/src/main/java/org/zowe/apiml/gateway/security/login/Providers.java#L36

Reply:
Thank you for providing details, I would change that.

Reply:
no problem, you can reach out to me via slack if you need something to discuss